addon-test-support/-private/patch-test-loader.js

```diff
@@ -33,14 +36,32 @@ export default function patchTestLoader(TestLoader) {
 
     const testLoader = this;
 
-    testLoader._testModules = [];
+    this.modules = testLoader._testModules = [];
     _super.loadModules.apply(testLoader, arguments);
 
-    const splitModules = splitTestModules(testLoader._testModules, split, partitions);
+    if (loadBalance) {
+      this.modules = weightTestModules(this.modules);
+    }
+
+    this.modules = splitTestModules(this.modules, split, partitions);
+
+    if (loadBalance) {
+      Testem.emit('set-modules-queue', this.modules);
```

Comment:
Why emit an event, rather then invoke a method?

Reply:
So essentially this is performing the communication between the browser and loader, and needs to send the modules through the socket to the browser to know what modules are to be queued, correct?

Reply:
It's sending the modules to server through the socket. server becomes a source of truth for the modules and browsers ask the server by emitting `get-next-module` event to get next module to run.

https://github.com/ember-cli/ember-exam/pull/136/files/c30c1e99e635910e7e21a01546534ae1a010a8e0#diff-babb5d4fe8ce7520c798eaae9f5393a0R54